instance_clusters.go

```diff
@@ -0,0 +1,206 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package gitlab
+
+import (
+	"fmt"
+	"time"
+)
+
+// InstanceClustersService handles communication with the
+// instance clusters related methods of the GitLab API.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ee/api/instance_clusters.html
+type InstanceClustersService struct {
+	client *Client
+}
+
+// InstanceCluster represents a GitLab Instance Cluster.
+//
+// GitLab API docs: https://docs.gitlab.com/ee/api/instance_clusters.html
+type InstanceCluster struct {
+	ID                 int                 `json:"id"`
+	Name               string              `json:"name"`
+	Domain             string              `json:"domain"`
+	CreatedAt          *time.Time          `json:"created_at"`
+	ProviderType       string              `json:"provider_type"`
+	PlatformType       string              `json:"platform_type"`
+	EnvironmentScope   string              `json:"environment_scope"`
+	ClusterType        string              `json:"cluster_type"`
+	User               *User               `json:"user"`
+	PlatformKubernetes *PlatformKubernetes `json:"platform_kubernetes"`
+	ManagementProject  *ManagementProject  `json:"management_project"`
+}
+
+func (v InstanceCluster) String() string {
+	return Stringify(v)
+}
+
+// PlatformKubernetes represents a GitLab Instance Cluster PlatformKubernetes.
+type InstanceClusterPlatformKubernetes struct {
+	APIURL            string `json:"api_url"`
+	Token             string `json:"token"`
+	CaCert            string `json:"ca_cert"`
+	Namespace         string `json:"namespace"`
+	AuthorizationType string `json:"authorization_type"`
+}
+
+// ListClusters gets a list of all instance clusters.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ee/api/instance_clusters.html#list-instance-clusters
+func (s *InstanceClustersService) ListClusters(options ...RequestOptionFunc) ([]*InstanceCluster, *Response, error) {
+	u := fmt.Sprintf("admin/clusters")
+
+	req, err := s.client.NewRequest("GET", u, nil, options)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var ics []*InstanceCluster
+	resp, err := s.client.Do(req, &ics)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return ics, resp, err
+}
+
+// GetCluster gets an instance cluster.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ee/api/instance_clusters.html#get-a-single-instance-cluster
+func (s *InstanceClustersService) GetCluster(clusterID int, options ...RequestOptionFunc) (*InstanceCluster, *Response, error) {
```

Comment:
For consistency, I prefer to call this param `cluster` instead of `clusterID` (while I agree this would probably be a good, or maybe even better, name as well).